packages/foundry/cli.ts

```diff
@@ -0,0 +1,69 @@
+import { Command, OptionValues } from "@commander-js/extra-typings";
+
+import { Options, TemplateInterface, Templates } from "./templates";
+
+import * as pkg from "./package.json";
+import process from "process";
+import path from "path";
+
+let outputDirectory = "";
+let repositoryUrl: string;
+
+export interface Configuration {
+  outputDirectory: string;
+  repositoryUrl: string;
+}
+
+const AddDefaultOptionsToCommand = (command: Command): void => {
+  command.option(
+    "-o, --out-dir <string>",
+    `where to create the template`,
+    process.cwd()
+  );
+};
+
+const AddCommand = (
+  program: Command,
+  name: string,
+  template: TemplateInterface
+) => {
+  const command = program.command(name);
+
+  if (template.description) {
+    command.description(template.description);
+  }
+
+  AddDefaultOptionsToCommand(command);
+
+  if (template.options && template.options.length > 0) {
+    for (const option of template.options) {
+      command.option(option.flag, option.description, option.defaultValue);
+    }
+  }
+
+  command.action((options: OptionValues, command) => {
+    const outDir = options["outDir"]?.toString() ?? process.cwd();
+    outputDirectory = path.resolve(outDir);
+
+    repositoryUrl = Templates[command.name()].repositoryUrl;
+
+    template.action ? template.action(options as Options) : options;
+  });
+};
+
+export const RunCli = (): Configuration => {
+  const program = new Command();
+
+  program.name(pkg.name).description(pkg.description).version(pkg.version);
```

Comment:
Using the package description could be a bad idea if we want to introduce rich markup along the way that might not be allowed in a `package.json` file.